### PR TITLE
[INLONG-5842][Manager] Support maintenance of message queue cluster

### LINKED ISSUE #5842: [Feature][Manager] Support maintenance of message queue cluster
### Description

1. Control produce operation and consume operation of Inlong message queue cluster.
2. Build relationships between DataProxy cluster and MessageQueue cluster.
3. Remove relationships between DataProxy cluster and MessageQueue cluster.
4. Synchronize all topic from cluster tag to message queue cluster.
5. Clear all topic from a message queue cluster.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Prepare a Pull Request
- Title: [INLONG-5842][Manager] Support maintenance of message queue cluster
- Fixes #5842 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+            return this.createTubeTopic(mqCluster, groupEntities);
+        } else {
+            return String.format("Unknown message queue type:%s", mqType);
+        }
+    }
+
+    /**
+     * createPulsarTopic
+     */
+    private String createPulsarTopic(InlongClusterEntity cluster, List<InlongGroupEntity> groupInfos) {
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:244]: Very famaliar logic with the **_createXXXXTopic_** in **ClusterTagService**. 
Can use **ClusterTagService** dicretly?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+    @Autowired
+    private InlongGroupEntityMapper groupMapper;
+    @Autowired
+    private TubeMQOperator tubeMQOperator;
+
+    /**
+     * Control produce operation and consume operation of Inlong message queue cluster 
+     */
+    @Override
+    public String control(MessageQueueControlRequest request) {
```
Comment (leosanqing) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:96]: Use ValidationUtils to valid fields, pls. You can refer to org.apache.inlong.manager.common.utils.ValidationUtilsTest#testValidate

But，first you should add annotations, such as @NotBlank,@Notnull on field,

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+    }
+
+    /**
+     * createKafkaTopic
+     */
+    private String clearKafkaTopic(InlongClusterEntity cluster) {
+        AdminClient admin = null;
+        java.lang.reflect.Type type = new TypeToken<HashMap<String, String>>() {
+        }.getType();
+        try {
```
Comment (leosanqing) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:524]: Use try-with-resources to close stream, pls

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+            Map<String, String> extParams = gson.fromJson(strExtParams, type);
+            Properties properties = new Properties();
+            properties.putAll(extParams);
+            // create admin
+            admin = AdminClient.create(properties);
+            // check topic exist
+            Set<String> topics = admin.listTopics().names().get();
+            DeleteTopicsResult result = admin.deleteTopics(topics);
+            result.all().get();
+        } catch (Exception e) {
```
Comment (leosanqing) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:536]: Just throw exception directly, framework will process. pls, refer to org.apache.inlong.manager.web.config.ControllerExceptionHandler#handleBusinessExceptionHandler

```suggestion
        } catch (Exception e) {
        ....
        throw new BusinessException(errorMsg);
```

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+@RequestMapping("/openapi")
+@Api(tags = "Message-Queue-API")
+public class MessageQueueController {
+
+    @Autowired
+    private MessageQueueService queueService;
+
+    @RequestMapping(value = "/cluster/queue/control", method = RequestMethod.POST)
+    @ApiOperation(value = "Control produce operation and consume operation of Inlong message queue cluster ")
+    public Response<String> control(@RequestBody MessageQueueControlRequest request) {
```
Comment (leosanqing) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java:49]: if Service layer throw exception, controller layer Just like this:
```suggestion
public Response<Void> control(@RequestBody MessageQueueControlRequest request) {
         queueService.control(request);
             return Response.success();
     }
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java (hunk 1 above)
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:244]: It is different, ClusterTagService create a topic for every MessageQueue cluster.
MessageQueueService create all topic of this cluster tag.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+    public String online(MessageQueueOnlineRequest request) {
+        String mqClusterName = request.getMqClusterName();
+        String proxyClusterName = request.getProxyClusterName();
+        return this.updateExtTag(mqClusterName, proxyClusterName, true);
+    }
+
+    /**
+     * Remove relationships between DataProxy cluster and MessageQueue cluster
+     */
+    public String offline(MessageQueueOfflineRequest request) {
```
[the remark this answers is not in the document]
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:143]: add @Override

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+ */
+@RestController
+@RequestMapping("/openapi")
+@Api(tags = "Message-Queue-API")
+public class MessageQueueController {
+
+    @Autowired
+    private MessageQueueService queueService;
+
+    @RequestMapping(value = "/cluster/queue/control", method = RequestMethod.POST)
```
[the remark this answers is not in the document]
Comment (luchunliang) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java:47]: change to @PostMapping

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java (hunk 5 above)
Comment (luchunliang) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java:49]: change to Void

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java (hunk 4 above)
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java:536]: change to BusinessException

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java (hunk 3 above)
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java]: change to  try-with-resources

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,555 @@
...
+            cluster.setExtTag(newExtTag);
+            clusterMapper.updateById(cluster);
+        }
+        return null;
+    }
+
+    /**
+     * Build relationships between DataProxy cluster and MessageQueue cluster
+     */
+    public String online(MessageQueueOnlineRequest request) {
```
[the remark this answers is not in the document]
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java]: fixed

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java (hunk 2 above)
Comment (luchunliang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/queue/MessageQueueServiceImpl.java]: change to ValidationUtils.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+@RequestMapping("/openapi")
+@Api(tags = "Message-Queue-API")
+public class MessageQueueController {
+
+    @Autowired
+    private MessageQueueService queueService;
+
+    @PostMapping("/cluster/queue/control")
+    @ApiOperation(value = "Control produce operation and consume operation of Inlong message queue cluster ")
+    public Response<Void> control(@RequestBody MessageQueueControlRequest request) {
```
Comment (leosanqing) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/MessageQueueController.java:49]: Pls add @Validated,if necessary

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/queue/MessageQueueOnlineRequest.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ApiModel("Build relationships between DataProxy cluster and MessageQueue cluster")
+public class MessageQueueOnlineRequest {
+
+    @NotBlank(groups = UpdateValidation.class, message = "miss message queue cluster name.")
```
Comment (leosanqing) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/queue/MessageQueueOnlineRequest.java:43]: If there is only one conditional check, only need this
```suggestion
    @NotBlank(message = "miss message queue cluster name.")
```

**Review by vernedeng — APPROVED** [review]
lgtm

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/queue/MessageQueueClearTopicRequest.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ApiModel("Clear all topic from a message queue cluster")
+public class MessageQueueClearTopicRequest {
+
+    @NotBlank(groups = UpdateValidation.class, message = "miss message queue cluster name.")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/cluster/queue/MessageQueueClearTopicRequest.java:43]: Suggested changing to `@NotBlank(message = "cluster name cannot be blank")`.

**Pull request closed without merging**